tools/protoxform/protoxform.py

```diff
@@ -27,6 +27,11 @@
 CLANG_FORMAT_STYLE = ('{ColumnLimit: 100, SpacesInContainerLiterals: false, '
                       'AllowShortFunctionsOnASingleLine: false}')
 
+NEXT_FREE_FIELD_MIN = 3
+NEXT_FREE_FIELD_ANNOTATION_FORMAT = " [#comment:next free field: %d]\n"
```

Comment:
I wonder if we can make this a a first class annotation in https://github.com/envoyproxy/envoy/blob/master/tools/api_proto_plugin/annotations.py, e.g. `[#next-free-field: 42]` and then have it parsed out via the existing machinery. Then we can add some generic machinery for doing annotations transforms, e.g. the type context can be given some lambda to transform different types of annotations.

Reply:
yeah, that is also exactly what in my mind.